src/libraries/System.Private.CoreLib/src/System/Threading/Interlocked.cs

```diff
@@ -67,9 +68,85 @@ public static partial class Interlocked
         /// <exception cref="NullReferenceException">The address of location1 is a null pointer.</exception>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static short Exchange(ref short location1, short value) =>
+            (short)Exchange(ref Unsafe.As<short, ushort>(ref location1), (ushort)value);
+
+        /// <summary>Sets a 8-bit unsigned integer to a specified value and returns the original value, as an atomic operation.</summary>
+        /// <param name="location1">The variable to set to the specified value.</param>
+        /// <param name="value">The value to which the <paramref name="location1"/> parameter is set.</param>
+        /// <returns>The original value of <paramref name="location1"/>.</returns>
+        /// <exception cref="NullReferenceException">The address of location1 is a null pointer.</exception>
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe byte Exchange(ref byte location1, byte value)
+        {
+#if !MONO && (TARGET_X86 || TARGET_AMD64 || TARGET_ARM64)
+            return Exchange(ref location1, value); // Must expand intrinsic
+#else
+            // this relies on GC keeping 4B alignment for refs and on subtracting to such alignment being in the same object
+            nuint offset = (nuint)Unsafe.AsPointer(ref location1) % sizeof(uint);
```

Comment:
We want to avoid direct `Unsafe.AsPointer` usage where-ever possible.

Instead, please add an internal named method (similar to https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/Unsafe.cs#L929) which makes it clear what's being done and why its "safe".

In this case, I think you can just use `OpportunisticMisalignment` directly, as that's exactly what you're computing here.